components/builder-worker/src/server.rs

```diff
@@ -177,17 +191,21 @@ impl Server {
     }
 
     fn set_busy(&mut self) -> Result<()> {
-        self.hb_cli.set_busy()?;
         self.state = State::Busy;
         Ok(())
     }
 
     fn set_ready(&mut self) -> Result<()> {
-        self.hb_cli.set_ready()?;
         self.state = State::Ready;
         Ok(())
     }
 
+    // Broadcast to subscribers the HeartbeatMgr health and state
+    fn send_heartbeat(&mut self) -> Result<()> {
+        self.fe_sock.send(&message::encode(&self.heartbeat)?, 0)?;
```

Comment:
do we need a separate function for this? I think we are just calling it once right?